pyang/statements.py

```diff
@@ -1814,20 +1830,22 @@ def add_tmp_children(node, tmp_children):
                         (stmt.arg, stmt.pos, c.arg, ch.pos))
                 return
         elif stmt.i_target_node.keyword == 'choice' and c.keyword != 'case':
-            # create an artificial case node for the shorthand
-            new_case = create_new_case(ctx, stmt.i_target_node, c, expand=False)
-            new_case.parent = stmt.i_target_node
-            v_inherit_properties(ctx, stmt.i_target_node, new_case)
+            if keyword_is_expected(stmt.pos, stmt.i_target_node, c) is True:
+                # create an artificial case node for the shorthand
+                new_case = create_new_case(ctx, stmt.i_target_node, c, expand=False)
+                new_case.parent = stmt.i_target_node
+                v_inherit_properties(ctx, stmt.i_target_node, new_case)
```

Comment:
Please add an  `else`  statement for the error handling. Although there may produce two `err_add`s, but it looks more reasonable.

Reply:
Sure. I will update it by your comments.